sonarts-core/src/rules/noRedundantJumpRule.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * SonarTS
+ * Copyright (C) 2017-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as tslint from "tslint";
+import * as ts from "typescript";
+import { SonarRuleMetaData } from "../sonarRule";
+import { SonarRuleVisitor } from "../utils/sonarAnalysis";
+import { ControlFlowGraph } from "../cfg/cfg";
+import { is, isBreakStatement, isContinueStatement, isReturnStatement } from "../utils/nodes";
+import { firstLocalAncestor } from "../utils/navigation";
+
+export class Rule extends tslint.Rules.AbstractRule {
+  public static metadata: SonarRuleMetaData = {
+    ruleName: "no-redundant-jump",
+    description: "Jump statements should not be redundant",
+    rationale: tslint.Utils.dedent``,
+    optionsDescription: "",
+    options: null,
+    rspecKey: "RSPEC-3626",
+    type: "functionality",
+    typescriptOnly: false,
+  };
+
+  public static MESSAGE = "Remove this redundant jump.";
+
+  public apply(sourceFile: ts.SourceFile): tslint.RuleFailure[] {
+    return new Visitor(this.getOptions().ruleName).visit(sourceFile).getIssues();
+  }
+}
+
+class Visitor extends SonarRuleVisitor {
+  public visitFunctionDeclaration(func: ts.FunctionDeclaration) {
+    if (!func.body) return;
+    const cfg = ControlFlowGraph.fromStatements(Array.from(func.body.statements));
+    if (cfg) {
+      for (let block of cfg.getBlocks()) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Identifier 'block' is never reassigned; use 'const' instead of 'let'. ([typescript:S3353](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS3353))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&pullRequest=628&issues=AWNE6bDBlRUrfRWxk6O5&open=AWNE6bDBlRUrfRWxk6O5)

[ISSUE_KEY]: AWNE6bDBlRUrfRWxk6O5